langchain_lakefs/document_loaders.py

```diff
@@ -1,72 +1,161 @@
-    """  # noqa: E501
-
-    # TODO: This method must be implemented to load documents.
-    # Do not implement load(), a default implementation is already available.
-    def lazy_load(self) -> Iterator[Document]:
-        raise NotImplementedError()
-
-    # TODO: Implement if you would like to change default BaseLoader implementation
-    # async def alazy_load(self) -> AsyncIterator[Document]:
+    def __init__(
+            self,
+            lakefs_endpoint: str,
+            lakefs_access_key: str,
+            lakefs_secret_key: str,
+            repo: str,
+            ref: str = "main",
+            path: str = "",
+    ):
+        self.client = Client(
+            host=lakefs_endpoint,
+            username=lakefs_access_key,
+            password=lakefs_secret_key
+        )
+        self.repo = repo
+        self.ref = ref
+        self.path = path
+
+    def set_path(self, path: str) -> None:
+        self.path = path
+
+    def set_ref(self, ref: str) -> None:
+        self.ref = ref
+
+    def set_repo(self, repo: str) -> None:
+        self.repo = repo
+
+    def set_user_metadata(self, user_metadata: bool) -> None:
+        self.user_metadata = user_metadata
+
+    def load(self) -> List[Document]:
+        """Load documents from lakeFS using presigned URLs if supported."""
+
+        self.__validate_instance()
+
+        presigned = self.client.storage_config.pre_sign_support
+        objects = lakefs.repository(self.repo, client =self.client).ref(self.ref).objects(user_metadata=True, prefix=self.path)
+        documents = [
+            doc
+            for obj in objects  # Iterate over ObjectInfo instances
+            for doc in UnstructuredLakeFSLoader(
+                obj.physical_address,  # Extract physical_address
+                self.repo,
+                self.ref,
+                obj.path,  # Extract path
+                presigned,
+                user_metadata=obj.metadata,  # Extract metadata
+                client=self.client,
+            ).load()
+        ]
+
+        return documents
+
+    def __validate_instance(self) -> None:
+        if self.repo is None or self.repo == "":
+            raise ValueError(
+                "no repository was provided. use `set_repo` to specify a repository"
+            )
+        if self.ref is None or self.ref == "":
+            raise ValueError("no ref was provided. use `set_ref` to specify a ref")
+        if self.path is None:
+            raise ValueError("no path was provided. use `set_path` to specify a path")
+
+
+class UnstructuredLakeFSLoader(UnstructuredBaseLoader):
+    """Load from `lakeFS` as unstructured data."""
+
+    def __init__(
+            self,
+            url: str,
+            repo: str,
+            ref: str = "main",
+            path: str = "",
+            presign: bool = True,
+            client: Optional[Client] = None,
+            # presign: bool = False,
+            user_metadata: Optional[dict] = None,
```

Comment:
remove the comment and try typed dict (will require import)
```suggestion
            user_metadata: Optional[Dict[str, str]] = None,
```